ci-operator/step-registry/cucushift/installer/check/gcp/user-tags/cucushift-installer-check-gcp-user-tags-commands.sh

```diff
@@ -29,46 +29,83 @@ then
 function validate_user_tags() {
   local -r current_tags="$1";  shift
 
-  local ret=0 cnt=1 a_tag_value
+  local cnt=1 a_tag_value
+  echo "" > "${validation_result_file}"
   printf '%s' "${USER_TAGS:-}" | while read -r PARENT KEY VALUE || [ -n "${PARENT}" ]
   do
     a_tag_value="namespacedTagValue: ${PARENT}/${KEY}/${VALUE}"
     if echo "${current_tags}" | grep -Fq "${a_tag_value}"; then
       echo "$(date -u --rfc-3339=seconds) - Found tag ${cnt} '${a_tag_value}' (PARENT/KEY/VALUE)."
       cnt=$(( $cnt + 1 ))
+      echo 0 >> "${validation_result_file}"
       continue
     else
       echo "$(date -u --rfc-3339=seconds) - Failed to find tag '${a_tag_value}' (PARENT/KEY/VALUE)."
-      ret=1
+      echo 1 >> "${validation_result_file}"
     fi
   done
-  return $ret
 }
 
 ## Try the validation
 ret=0
-validation_ret=0
+
+echo "$(date -u --rfc-3339=seconds) - Checking userTags of machines..."
+readarray -t items < <(gcloud compute instances list --filter="name~${CLUSTER_NAME}" --format="table(name,zone)" | grep -v NAME)
+for line in "${items[@]}"; do
+  name="${line%% *}"
+  zone="${line##* }"
+  current_tags="$(gcloud resource-manager tags bindings list --parent=//compute.googleapis.com/projects/${GOOGLE_PROJECT_ID}/zones/${zone}/instances/${name} --location=${zone} --effective)"
+  echo "${current_tags}"
+  validate_user_tags "${current_tags}"
+  if grep -q "1" "${validation_result_file}"; then
+    echo "$(date -u --rfc-3339=seconds) - FAILED for machine '${name}'."
+    ret=1
+  else
+    echo "$(date -u --rfc-3339=seconds) - PASSED for machine '${name}'."
+  fi
+done
+
+echo "$(date -u --rfc-3339=seconds) - Checking userTags of disks..."
+readarray -t items < <(gcloud compute disks list --filter="name~${CLUSTER_NAME}" --format="table(name,zone)" | grep -v NAME)
+for line in "${items[@]}"; do
+  name="${line%% *}"
+  zone="${line##* }"
+  zone=$(basename ${zone})
+  disk_id=$(gcloud compute disks describe ${name} --zone ${zone} --format json | jq -r -c .id)
+  current_tags="$(gcloud resource-manager tags bindings list --parent=//compute.googleapis.com/projects/${GOOGLE_PROJECT_ID}/zones/${zone}/disks/${disk_id} --location=${zone} --effective)"
+  echo "${current_tags}"
+  validate_user_tags "${current_tags}"
+  if grep -q "1" "${validation_result_file}"; then
+    echo "$(date -u --rfc-3339=seconds) - FAILED for disk '${name}'."
+    ret=1
+  else
+    echo "$(date -u --rfc-3339=seconds) - PASSED for disk '${name}'."
+  fi
+done
 
 echo "$(date -u --rfc-3339=seconds) - Checking userTags of image-registry buckets..."
 readarray -t items < <(gsutil ls | grep "${INFRA_ID}-image-registry")
 for line in "${items[@]}"; do
   name=$(basename "${line}")
   current_tags="$(gcloud resource-manager tags bindings list --parent=//storage.googleapis.com/projects/_/buckets/${name} --location=${GCP_REGION} --effective)"
   echo "${current_tags}"
-  validate_user_tags "${current_tags}" || validation_ret=$?
```

Comment:
I wonder why we changed the checking way of `validate_user_tags` by return value?

Reply:
I happened noticing the previous way not working, i.e. even I set "ret=1" for a failure within the "while" loop, the function will still return "0". Not quite sure why. Anyway, the new way does work for both failure and success scenarios. Any suggestion or concern? Thanks!

Reply:
Just be curious why the original way does not work, generally we can `set -x` for debugging to locate the issue. I have no objection on the new change, keep it as your desire.

Reply:
Got it, let me try some time later. Thanks!